### PR TITLE
Sign up test

### PR DESCRIPTION
**Pull request by cristianames92**
## Summary

Adds tests to the feature of sign up

## Known Issues

none

## Trello Card

https://trello.com/c/hNk0leXx/33-tests-y-documentaci%C3%B3n-de-sign-up

### CODE REVIEW

File: test/app.js [hunk 1]
```diff
@@ -9,11 +9,16 @@ const fs = require('fs'),
...
-const getTablesQuery = `SELECT table_name FROM information_schema.tables WHERE table_schema = 'public' AND table_type = 'BASE TABLE';`;
+// const getTablesQuery = `SELECT table_name FROM information_schema.tables WHERE table_schema = 'public' AND table_type = 'BASE TABLE';`;
+
+const getTablesQuery = `SELECT table_name FROM information_schema.tables WHERE table_schema = 'public' AND table_type = 'BASE TABLE' AND table_name <> 'SequelizeMeta';`;
 
 // THIS WORKS ONLY WITH POSTGRESQL
 beforeEach('drop tables, re-create them and populate sample data', done => {
   models.sequelize.query(getTablesQuery).then(tables => {
+    if (!tables.length) {
+      return done();
```
Comment (gonziesc) [inline test/app.js]: just done

File: test/controllers/users.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    };
+
+    context('When creating a valid user', () => {
+      it('returns status 201', done => {
+        chai
+          .request(server)
+          .post(NEW_USER_ENDPOINT)
+          .send(params)
+          .then(res => {
+            res.should.have.status(201);
```
Comment (gonziesc) [inline test/controllers/users.js:46]: lets use expect instead of should

File: test/controllers/users.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          first_name: 'Cristian'
+        };
+
+        it('returns status 500', done => {
+          chai
+            .request(server)
+            .post(NEW_USER_ENDPOINT)
+            .send(invalidParams)
+            .catch(res => {
+              res.status.should.be.eq(500);
```
Comment (gonziesc) [inline test/controllers/users.js:112]: it should return 500 if missing params? with not 422?

**Review by gonziesc — CHANGES_REQUESTED** [review]

File: test/controllers/users.js (hunk 3 above)
Comment (cristianames92) [reply test/controllers/users.js:112]: I didn't want to spend time creating more types of errors, knowing that a 500 is not precisely a user error.

File: test/app.js (hunk 1 above)
Comment (cristianames92) [reply test/app.js]: 👍

File: test/controllers/users.js (hunk 2 above)
Comment (cristianames92) [reply test/controllers/users.js:46]: Good to know. I already created all the subsequent tests using should. Can we avoid making this refactor for now?

Comment (gonziesc) [reply test/controllers/users.js:46]: no problem

File: test/controllers/users.js (hunk 3 above)
Comment (gonziesc) [reply test/controllers/users.js:112]: ok

**Review by gonziesc — COMMENTED** [review]
Code is ok, what about migrations? or aren't you doing that part?

**Comment by cristianames92** [comment]
@gonziesc already in master for some reason
https://github.com/wolox-training/ca-express-js/blob/master/migrations/migrations/20180709153151-create_user_model.js

**Review by gonziesc — APPROVED** [review]

**Pull request merged**